### ISSUE TITLE
[VS Code] Client cannot establish websocket connection with web worker

### ISSUE THREAD

**Issue by kmelmon** [microsoft/react-native-windows#2877]
Currently, when using the VS Code debugging extension for react-native, starting a debugging session is not working. This issue is tracking one of the issues in that experience.

This issue is first dependent upon this issue being fixed:
https://github.com/microsoft/react-native-windows/issues/2876

Once that is fixed, you'll run into the second issue, which this issue is tracking.  The issue here is  that when the client tries to make a websocket connection with the remote debugger process, it fails.  The issue isn't understood yet, but my instinct is it's related to how the VS Code extension starts up the web worker that hosts the javascript which will be debugged.

**Comment by mganandraj** [comment]
I was and am doing some work related to this. I'm mostly focusing on in-proc debugging though, i.e. not loading JS into an Chrome browser tab. As part of the Hermes integration, i'll try to bring up directly debugging the Hermes runtime from chrome. A few months back, I did a similar work with v8 which also i'll try to get in. Both requires setting up a websocket server in-app so that the clients such as vscode and chrome inspector can interact. When working with v8, I had faced connectivity issues when the websocket stack in chrome couldn't create a stable connection iwht boost/beast based websocket server created in-app. eventually, I ended up copying the websocket stack from chrome browser to our app, and it worked. Not sure whether the connectivity issue that your are seeing, is releated.
